pkg/security/ptracer/cws.go

```diff
@@ -586,15 +808,33 @@ func StartCWSPtracer(args []string, envs []string, probeAddr string, creds Creds
 					logErrorf("unable to handle fchdir: %v", err)
 					return
 				}
+			case CloseNr:
+				if err = handleClose(tracer, process, syscallMsg, regs); err != nil {
```

Comment:
Note sure about this but maybe we should remove the fd from the cache in the `Post` callback to make sure the `close` syscall was sucecssful.

Reply:
Yeah I also wonder. My call is that, there is not many error cases: either the provided fd is not valid (so, deleting it wont hurt anyway), I/O error (I'm not sure how it can happens on a close, but then the FD should be ~dead too), or interrupted by a signal. In any case, for caller once the close() is called, IMO it won't/cant use anymore the given fd. So it's simpler to do the cleanup on the PRE instead of the POST (avoiding to add another fake msg struct to cache the fd until the POST).
If you have a different takes please tell me :) happy to discuss and change to a post cleanup if needed :)